tests/Application/Actions/RequestPasswordResetTest.php

```diff
@@ -0,0 +1,114 @@
+        $personId = Uuid::v4();
+        $emailAddress = 'donor@weliketodonatebutweforgotpasswords.com';
+
+        // arrange
+        $person = new Person();
+        $person->setId($personId);
+
+        $personRepoProphecy->findPasswordEnabledPersonByEmailAddress($emailAddress)
+            ->willReturn($person);
+
+        $passwordResetTokenProphecy = $this->prophesize(PasswordResetTokenRepository::class);
+
+        $mailerProphecy = $this->prophesize(Mailer::class);
+        $mailerProphecy->sendEmail(Argument::any())->willReturn(true);
+
+        $container = $app->getContainer();
+        assert($container instanceof Container);
+        $container->set(PasswordResetTokenRepository::class, $passwordResetTokenProphecy->reveal());
+        $container->set(PersonRepository::class, $personRepoProphecy->reveal());
+        $container->set(Mailer::class, $mailerProphecy->reveal());
+
+        // asert
+
+        $passwordResetTokenProphecy->persist(Argument::that(function (PasswordResetToken $token) use ($person){
+            return $token->person === $person;
+        }))->shouldBeCalledOnce();
+
+        // act
+        $app->handle($this->buildRequest([
+            'email_address' => $emailAddress,
+        ]));
+    }
+
+    public function testRequestingPasswordResetSendsEmail(): void
+    {
+        $app = $this->getAppInstance();
+        $personRepoProphecy = $this->prophesize(PersonRepository::class);
+        $personId = Uuid::v4();
+        $emailAddress = 'donor@weliketodonatebutweforgotpasswords.com';
+
+        // arrange
+        $person = new Person();
+        $person->setId($personId);
+        $person->email_address = $emailAddress;
+        $person->first_name = "Joe";
+        $person->last_name = "Bloggs";
+
+        $personRepoProphecy->findPasswordEnabledPersonByEmailAddress($emailAddress)
+            ->willReturn($person);
+
+        $passwordResetTokenProphecy = $this->prophesize(PasswordResetTokenRepository::class);
+        $container = $app->getContainer();
+        assert($container instanceof Container);
+        $container->set(PasswordResetTokenRepository::class, $passwordResetTokenProphecy->reveal());
+        $container->set(PersonRepository::class, $personRepoProphecy->reveal());
+
+        // assert
+        $mailerProphecy = $this->prophesize(Mailer::class);
+        $container->set(Mailer::class, $mailerProphecy->reveal());
+
+        $mailerProphecy->sendEmail(Argument::that(function (array $params) use ($emailAddress) {
+            /** @var array<string, string> $params */
+            $this->assertSame('password-reset-requested', $params['templateKey']);
+            $this->assertSame($emailAddress, $params['recipientEmailAddress']);
+            $this->assertMatchesRegularExpression('/http.*/', $params['params']['resetLink']);
+            $this->assertSame('Joe', $params['params']['firstName']);
+            $this->assertSame('Bloggs', $params['params']['lastName']);
+
+            return true;
```

Comment:
@bdsl I might be mistaken, but do we need this line if we already check `->willReturn(true)` in the next line?

Reply:
Yes we do. It's the return from the closure used in the test to say that the function call meets expectations, not the return from the actual test double. If I delete it and run tests we get an error:

```

1) BigGive\Identity\Tests\Application\Actions\RequestPasswordResetTest::testRequestingPasswordResetSendsEmail
TypeError: Double\BigGive\Identity\Client\Mailer\P9::sendEmail(): Return value must be of type bool, null returned

/var/www/html/src/Application/Actions/CreatePasswordResetToken.php:62
```